### PR TITLE
[14.0][IMP] mrp_subcontracting_inhibit: Add tests to check the behaviour of subcontracting_inhibit

### PR DESCRIPTION
**Pull request by victoralmau**
Add tests to check the behaviour of `subcontracting_inhibit`

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT38580

### CODE REVIEW

File: mrp_subcontracting_inhibit/models/purchase_order.py [hunk 1]
```diff
@@ -27,6 +27,10 @@ def _prepare_purchase_order_line_from_procurement(
         res.update({"subcontracting_inhibit": subcontracting_inhibit_value})
         return res
 
+    @api.onchange("subcontracting_inhibit")
+    def _onchange_subcontracting_inhibit(self):
+        self._onchange_quantity()
```
Comment (pedrobaeza) [inline mrp_subcontracting_inhibit/models/purchase_order.py]: Explain why this with the docstring

File: mrp_subcontracting_inhibit/tests/test_mrp_subcontracting_inhibit.py [hunk 2]
```diff
@@ -100,3 +100,13 @@ def test_misc_buy_subcontractor_route(self):
         self.assertEqual(order.order_line.price_unit, 5)
         order.button_confirm()
         self.assertEqual(self._get_mrp_production_total(), 0)
+
+    def test_purchase_order(self):
```
Comment (pedrobaeza) [inline mrp_subcontracting_inhibit/tests/test_mrp_subcontracting_inhibit.py]: Please use a more specific method name

**Comment by victoralmau** [comment]
Changes done.

**Review by pedrobaeza — APPROVED** [review]

**Comment by victoralmau** [comment]
Ping @chienandalu

**Comment by pedrobaeza** [comment]
@victoralmau can you rebase and fix conflicts?

**Comment by victoralmau** [comment]
Rebase done.

**Comment by pedrobaeza** [comment]
It's weird that on Odoo it gives error, but not on OCB. Maybe a change that hasn't arrived on OCB?

**Review by chienandalu — APPROVED** [review]

**Pull request merged**